app/main.py

```diff
@@ -64,27 +88,29 @@ def make_middleware() -> List[Middleware]:
         Middleware(
             EventHandlerASGIMiddleware,
             handlers=[local_handler]
+        ),
+        Middleware(
+            LogMiddleware, 
+            logger=uvicorn_logger,
+            file_logger=file_logger
         )
     ]
 
+app = FastAPI(
+    openapi_url=f"{ settings.API_V1_STR }/openapi.json",
+    middleware=make_middleware()
+)
 
-logger = logging.getLogger(__name__)
-config_path=Path(__file__).with_name("logging_config.json")
-
-def create_app() -> FastAPI:
-    app = FastAPI(
-        openapi_url=f"{ settings.API_V1_STR }/openapi.json",
-        middleware=make_middleware()
+@app.exception_handler(CustomException)
+async def custom_exception_handler(request: Request, exc: CustomException):
+    content = { "error_code": exc.error_code, "message": exc.message }
+    if settings.DEV_PHASE == DevPhase.DEV:
+        content["stack"] = exc.stack
+    return JSONResponse(
+        status_code=exc.code,
+        content=content,
     )
-    if not settings.DISABLE_LOGGER:
-        logger = CustomizeLogger.make_logger(config_path)
-        app.logger = logger
-        app.add_middleware(LogMiddleware)
-    init_monkeypatch()
-    init_routers(app)
-    init_listeners(app)
-    add_pagination(app)
-    
-    return app
 
-app = create_app()
+init_monkeypatch()
+init_routers(app)
+add_pagination(app)
```

Comment:
why ?

Reply:
Why put all this in a method and call that method right afterward instead of just doing it at the root level? I just thought this made more sense, wasn't sure why it was done the other way.

Reply:
The purpose really just boils down to organization. Particularly in the above where init_listeners was removed, which in the job queue PR adds multiple new listener signals to the app lifecycle.

Reply:
It's fine. I don't see any particular reason one way or another at the moment. If we find good reason to change it back, it's simple enough to do so at that point.